sigmf/archivereader.py

```diff
@@ -7,86 +7,114 @@
-                    json_contents = memb.name
-                    if data_offset_size is None:
-                        # consider a warnings.warn() here; the datafile should be earlier in the
-                        # archive than the metadata, so that updating it (like, adding an annotation)
-                        # is fast.
-                        pass
-                    with tar_obj.extractfile(memb) as memb_fid:
-                        json_contents = memb_fid.read()
-
-                elif memb.name.endswith(SIGMF_DATASET_EXT):
-                    data_offset_size = memb.offset_data, memb.size
+    def __init__(self,
+                 path=None,
+                 skip_checksum=False,
+                 map_readonly=True,
+                 archive_buffer=None):
+        self.path = path
+        tar_obj = None
+        try:
+            if self.path is not None:
+                if not self.path.endswith(SIGMF_ARCHIVE_EXT):
+                    err = "archive extension != {}".format(SIGMF_ARCHIVE_EXT)
+                    raise SigMFFileError(err)
+
+                tar_obj = tarfile.open(self.path)
+
+            elif archive_buffer is not None:
+                tar_obj = tarfile.open(fileobj=archive_buffer, mode='r:')
 
-                else:
-                    print('A regular file', memb.name, 'was found but ignored in the archive')
             else:
-                print('A member of type', memb.type, 'and name', memb.name, 'was found but not handled, just FYI.')
-
-        if data_offset_size is None:
-            raise SigMFFileError('No .sigmf-data file found in archive!')
-
-        self.sigmffile = SigMFFile(metadata=json_contents)
-        valid_md = self.sigmffile.validate()
-
-        self.sigmffile.set_data_file(self.name, data_buffer=archive_buffer, skip_checksum=skip_checksum, offset=data_offset_size[0],
-                                     size_bytes=data_offset_size[1], map_readonly=map_readonly)
-
-        self.ndim = self.sigmffile.ndim
-        self.shape = self.sigmffile.shape
-
-        tar_obj.close()
+                raise ValueError('In sigmf.archivereader.__init__(), either '
+                                 '`path` or `archive_buffer` must be not None')
+
+            json_contents = None
+            data_offset_size = None
+            sigmffile_name = None
+            self.sigmffiles = []
+            data_found = False
+
+            for memb in tar_obj.getmembers():
+                if memb.isdir():  # memb.type == tarfile.DIRTYPE:
+                    # the directory structure will be reflected in the member
+                    # name
+                    continue
+
+                elif memb.isfile():  # memb.type == tarfile.REGTYPE:
+                    if memb.name.endswith(SIGMF_METADATA_EXT):
+                        json_contents = memb.name
+                        if data_offset_size is None:
+                            # consider a warnings.warn() here; the datafile
```

Comment:
I like the idea of adding a warning here as noted

Reply:
Removed this in [91e577f](https://github.com/NTIA/sigmf-python/pull/2/commits/91e577f6b69a47d4e5924e4e5fcb033a01398c8f) as seems no longer relevant